### PR TITLE
Config parser cleanups

### PR DESCRIPTION
**Pull request by pks-t**
Two small cleanups to make code more readable in our config parser.

### CODE REVIEW

File: src/config_file.c [hunk 1]
```diff
@@ -1056,36 +1056,38 @@ static int parse_conditional_include(git_config_parser *reader,
...
-	git_buf_printf(&buf, "%s.%s", current_section, var_name);
-	git__free(var_name);
+	name = git__strdup(var_name);
+	GITERR_CHECK_ALLOC(name);
 
-	if (git_buf_oom(&buf)) {
-		git__free(var_value);
+	git__strtolower(name);
+	git_buf_printf(&buf, "%s.%s", current_section, name);
+	git__free(name);
```
Comment (ethomson) [inline src/config_file.c]: It seems disappointing to have to do an extra tiny `malloc` here for `name` n every variable that we read and immediately `free` it.  Doing a little bit more manipulation and not doing a `printf` here might help.  eg:

```
git_buf_puts(&buf, current_section);
git_buf_putc(&buf, '.');

for (c = name, *c; c++)
    git_buf_putc(&buf, git__tolower(*c));
```

Comment (pks-t) [reply src/config_file.c]: Agreed. Your solution looks a lot cleaner, thanks!

**Comment by pks-t** [comment]
Improved as to @ethomson's proposal

File: src/config_file.c [hunk 2]
```diff
@@ -1056,26 +1056,28 @@ static int parse_conditional_include(git_config_parser *reader,
...
-	git_buf_printf(&buf, "%s.%s", current_section, var_name);
-	git__free(var_name);
+	git_buf_puts(&buf, current_section);
+	git_buf_putc(&buf, '.');
+	for (c = var_name; *c; c++)
+		git_buf_putc(&buf, git__tolower(*c));
 
 	if (git_buf_oom(&buf)) {
-		git__free(var_value);
+		git_buf_free(&buf);
```
Comment (carlosmn) [inline src/config_file.c]: There's no need to free an OOM'd buffer. If the realloc fails, we free the buffer we had before setting the buffer to the OOM one.

Comment (pks-t) [reply src/config_file.c]: Yeah, obviously. I've created a new commit for that, as there is also a second occasion where we had the same pattern nearby. Thanks for pointing out.

**Comment by pks-t** [comment]
Rebased to fix conflicts

**Comment by pks-t** [comment]
Rebased again to fix conflicts

**Comment by ethomson** [comment]
🎉

**Pull request merged**